### PR TITLE
Code Clean Up 

### PR DESCRIPTION
**Pull request by Pijukai**
## Content of the PR

This PR is doing a big cleanup of the code.

## Frequent modifications 

Some modifications are : 

- Add documentation 
- Remove `"@Suppress"` that erases errors 
- Remove Duplicate tests 
- Remove unused val and var 
- Remove unused arguments
- Restore naming conventions 
- Remove unused functions

## Origin of the errors 

Those are mainly errors that comes from warning of the compiler and the static analysis of the IDE. 

## Optimize imports

A big part of the added code is due to the optimization of imports I ran on the project. 

## Testing

I recommend that the reviewers test the app as deep as possible to verify none of the changes I made are creating bugs or removing fixes of bugs.

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ThemeTest.kt [hunk 1]
```diff
@@ -21,22 +20,21 @@ class ThemeTest {
...
   @Test
   fun disabledColorTestLight() {
-    composeTestRule.setContent { ColoredSquare(disabledColor(), LightColorScheme) }
+    composeTestRule.setContent { ColoredSquare(disabledColor()) }
     composeTestRule.onNodeWithTag("TestTag").assertIsDisplayed()
     // No test for checking the color
   }
 
   @Test
   fun disableColorTestDark() {
```
Comment (matttsss) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ThemeTest.kt:36]: Quick comment, there is another duplicated test here that you could delete!

Comment (andrepeiry) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ThemeTest.kt]: I was passing by, but could you check if you can remove the whole "part of color testing ? I never knew if it was really counted toward the code coverage ...

File: app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt [hunk 2]
```diff
@@ -364,11 +364,12 @@ fun SearchBarListScreen(
...
   val textFieldSize = remember { mutableStateOf(IntSize.Zero) }
 
-  val alpha by animateFloatAsState(targetValue = if (isTextFieldVisible.value) 1f else 0f)
+  val alpha by
+      animateFloatAsState(targetValue = if (isTextFieldVisible.value) 1f else 0f, label = "")
 
   // Animation values
   val slideOffset by
-      animateDpAsState(targetValue = if (isTextFieldVisible.value) 0.dp else (-200).dp)
+      animateDpAsState(targetValue = if (isTextFieldVisible.value) 0.dp else (-200).dp, label = "")
```
Comment (matttsss) [inline app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt:372]: Why did you add a label argument here? What was the warning there?

Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt:372]: Yes there was so I just added it. Do you think it is something I should remove ?

File: app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt [hunk 3]
```diff
@@ -472,16 +470,13 @@ fun AdminScreen(
                                               style = MaterialTheme.typography.titleMedium)
                                         },
                                         text = {
-                                          val selectedObject =
-                                              reportedObjectViewModel.selectedObject
-                                                  .collectAsState()
-                                                  .value
+                                          reportedObjectViewModel.selectedObject.collectAsState()
```
Comment (matttsss) [inline app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt:473]: I am not quite sure why we keep this call to collect as state when we don't assign it to a variable?

File: app/src/androidTest/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreenAlertDialogTest.kt [hunk 4]
```diff
@@ -98,7 +90,7 @@ class ParkingDetailsScreenAlertDialogTest {
       ParkingDetailsAlertDialogShowImage(
           parkingViewModel,
           userViewModel,
-          onDismiss = { dismissedCalled = true },
+          onDismiss = {},
```
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreenAlertDialogTest.kt:93]: why was this removed?

Comment (Pijukai) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreenAlertDialogTest.kt:93]: I removed `dismissedCalled` a few line above because it was not used in the test so it was useless. This var was never asserted.

File: app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt (hunk 3 above)
Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt:473]: I will remove it in the next commit I will just wait for you to finish your review so that I can address all your comments without hindering you on your review.

File: app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt [hunk 5]
```diff
@@ -472,16 +470,13 @@ fun AdminScreen(
...
                                           Column {
                                             if (selectedObject != null) {
                                               ReportDetailsContent(
-                                                  selectedObject.objectType,
-                                                  selectedObject.objectUID,
-                                                  selectedObject.userUID,
+                                                  selectedObject!!.objectType,
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt:476]: why do the !! re-become necessary? I specifically changed this yesterday especially because I enveloped it in a null check at @matttsss's request

**Review by LouisJamesVasseur — APPROVED** [review]
Some really great work all around. You clearly took time and effort to make our code cleaner and I appreciate it!

File: app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt (hunk 5 above)
Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt:476]: The IDE wouldn't let me use anything else. The null check is still here but you initialized `selectedObject` three times where it was not necessary so I removed everything to keep only the first declaration of the val. This way the IDE requested that I put !! here.

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/report/AdminScreen.kt:476]: Ah, okay. I see. Glad you found an even more efficient way to clean up the code!

**Review by matttsss — APPROVED** [review]
All good for me, nice to see the clear-up of the ViewModels!

**Comment by Pijukai** [comment]
Thank you for your reactivity :rocket:
